ddl/column.go

```diff
@@ -810,12 +811,20 @@ func (w *worker) doModifyColumnTypeWithData(
 			}
 		}
 		// delete only -> write only
-		updateChangingInfo(job, changingCol, changingIdxs, model.StateWriteOnly)
+		updateChangingInfo(changingCol, changingIdxs, model.StateWriteOnly)
 		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != changingCol.State)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+		job.SchemaState = model.StateWriteOnly
 	case model.StateWriteOnly:
 		// write only -> reorganization
-		updateChangingInfo(job, changingCol, changingIdxs, model.StateWriteReorganization)
+		updateChangingInfo(changingCol, changingIdxs, model.StateWriteReorganization)
 		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != changingCol.State)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+		job.SchemaState = model.StateWriteReorganization
```

Comment:
Do we need to change it in other operations like `onCreateIndex`?

Reply:
yes, I've added that in `onCreateIndex`, for all rest, I will file a new issue.